packages/@aws-cdk/aws-vpclattice-alpha/lib/service-network.ts

```diff
@@ -0,0 +1,150 @@
+  readonly services?: IService[];
+}
+
+/**
+ * Attributes for importing an existing VPC Lattice service network
+ */
+export interface ServiceNetworkAttributes {
+  /**
+   * The ARN of the service network
+   */
+  readonly serviceNetworkArn: string;
+
+  // TODO: restore ID? Constructing the ARN from the ID will need accountId, region, etc
+  // readonly serviceNetworkId?: string;
+}
+
+/**
+ * A VPC Lattice service network
+ */
+export class ServiceNetwork extends ServiceNetworkBase {
+  /**
+   * Import an existing VPC Lattice service network from its ARN
+   *
+   * @param scope The parent construct
+   * @param id The construct ID
+   * @param serviceNetworkArn The ARN of the service network
+   * @returns A service network construct
+   */
+  public static fromServiceNetworkArn(scope: Construct, id: string, serviceNetworkArn: string): IServiceNetwork {
+    return ServiceNetwork.fromServiceNetworkAttributes(scope, id, { serviceNetworkArn: serviceNetworkArn });
+  }
+
+  /**
+   * Import an existing VPC Lattice service network from its attributes
+   *
+   * @param scope The parent construct
+   * @param id The construct ID
+   * @param attrs The service network attributes
+   * @returns A service network construct
+   */
+  public static fromServiceNetworkAttributes(scope: Construct, id: string, attrs: ServiceNetworkAttributes): IServiceNetwork {
+    class UnownedServiceNetworkReference extends ServiceNetworkBase {
+      public readonly serviceNetworkArn: string = attrs.serviceNetworkArn;
+      // Extract the serviceNetworkId from the ARN, which has the format `arn:aws:vpc-lattice:<region>:<account>:servicenetwork/<service-network-id>`
+      public readonly serviceNetworkId: string = Stack.of(scope).splitArn(attrs.serviceNetworkArn, ArnFormat.SLASH_RESOURCE_NAME).resourceName!;
+    }
+    return new UnownedServiceNetworkReference(scope, id);
+  }
+
+  /**
+   * The ARN of the service network
+   * @attribute
+   * @returns a token representing the service network ARN
+   */
+  public readonly serviceNetworkArn: string;
+
+  /**
+   * The ID of the service network
+   * @attribute
+   * @returns a token representing the service network ID
+   */
+  public readonly serviceNetworkId: string;
+
+  /**
+   * Creates a new VPC Lattice service network
+   *
+   * @param scope The parent construct
+   * @param id The construct ID
+   * @param props The service network properties
+   */
+  constructor(scope: Construct, id: string, props: ServiceNetworkProps) {
+    super(scope, id);
+
+    // TODO: validation?
+
+    // TODO: tags
+
+    const resource = new CfnServiceNetwork(this, id, {
+      authType: props.authType,
+      name: props.serviceNetworkName,
+      sharingConfig: {
+        enabled: props.enableSharing ?? false,
+      },
+    });
+
+    this.serviceNetworkArn = resource.attrArn;
+    this.serviceNetworkId = resource.attrId;
+
+    // TODO: enforce at least one service? A service network without services is just...a network
```

Comment:
@mrgrain thoughts on this? IMO enforcing this is good, because a service network should have at least one service. However, this is not enforced by CFN or the AWS Console UX when creating a service network.

https://docs.aws.amazon.com/vpc-lattice/latest/ug/create-service-network.html

Reply:
> would prevent common coding patterns like

Agree with this, but the counter-argument would be: does a *service network* without at least one service (even briefly) make sense? 

I'm trying to think of use cases where the user's code would need to do this:
```
const sn = new ServiceNetwork(scope, 'ServiceNetwork', { ... });
```

and then use `sn` before associating a service.